### PR TITLE
Prevent accidental removal of essential trailing whitespaces in a test

### PR DESCRIPTION
**Pull request by MSeifert04**
I've seen several PRs that accidentally deleted the trailing whitespaces in the test. So I thought it may be good to have a test for the correctness of the test, so it will continue to test that trailing whitespaces don't mess with the reader.

To catch these accidental deletions I added an `assert` for the length of the `table`.

### CODE REVIEW

**Comment by MSeifert04** [comment]
CircleCI seems broken and I manually cancelled OSX on Travis.

**Comment by MSeifert04** [comment]
@astrofrog Are the two `**` at the end of the bot message intentional? I can't remember seeing the before.

**Comment by pllim** [comment]
The "**" is my bad. Just pretend they are not there!  ;)

**Comment by MSeifert04** [comment]
I like it when the machines show weaknesses. That shows they are still vulnerable.

On a more serious note: Who's going to review this? 😄

File: astropy/io/ascii/tests/test_rst.py [hunk 1]
```diff
@@ -147,6 +147,9 @@ def test_trailing_spaces_in_row_definition():
     1    4.5  bar
    ==== ==== ====  
 """  # noqa
+    # make sure noone accidentally deletes the trailing whitespaces in the
```
Comment (pllim) [inline astropy/io/ascii/tests/test_rst.py]: Nitpick: "noone" --> "no one"

**Comment by pllim** [comment]
> CircleCI seems broken

So it does. Will a rebase (if applicable) help?

**Comment by MSeifert04** [comment]
> Will a rebase (if applicable) help?

I don't know but I rebased and force-pushed. Let's see.

**Comment by MSeifert04** [comment]
Yay, CircleCI passes. 🎉

File: astropy/io/ascii/tests/test_rst.py [hunk 2]
```diff
@@ -147,6 +147,10 @@ def test_trailing_spaces_in_row_definition():
     1    4.5  bar
    ==== ==== ====  
 """  # noqa
+    # make sure no one accidentally deletes the trailing whitespaces in the
```
Comment (mhvk) [inline astropy/io/ascii/tests/test_rst.py]: While we're add this, could we perhaps ensure that auto-trailing-space deleting editors do not mess up the example? E.g., the following make the same input, but have quote marks at the end (also removing the need for `noqa`):
```
    table = ("""\n"""
"""# comment (with blank line above)\n"""
"""   ==== ==== ====    \n"""
"""   Col1 Col2 Col3\n"""
"""   ==== ==== ====  \n"""
"""    3    3.4  foo\n"""
"""    1    4.5  bar\n"""
"""   ==== ==== ====  \n""")
```
or perhaps better
```
    table = "\n".join([
"",
"# comment (with blank line above)",
"   ==== ==== ====    ",
"   Col1 Col2 Col3",
"   ==== ==== ====  ",
"    3    3.4  foo",
"    1    4.5  bar",
"   ==== ==== ====  ",
""])
```

**Comment by mhvk** [comment]
@MSeifert04 - looks OK, except I'd like to just remove the problem - maybe you can simply pick whichever suggestion you like best?

**Comment by MSeifert04** [comment]
@mhvk Does this look better? I don't like triple quotes for one-line strings. 😄

**Comment by bsipocz** [comment]
@MSeifert04 - I being one having an editor that auto removes the whitespaces, can't even say how much I like and appreciate this PR. Thanks!

**Comment by MSeifert04** [comment]
Yeah, AppVeyor and CircleCI passed the test so it's almost a safe bet Travis and OSX would've passed as well.

Thanks for all the feedback. 👍

**Pull request merged**